Clippy fixes

dashea opened the pull request:
Fix a bunch of warnings from clippy.

This uses a commit from https://github.com/weldr/bdcs-api-rs/pull/48/ for the needless-return warnings, and a commit from https://github.com/weldr/bdcs-api-rs/pull/45 for various other build warnings.

atodorov commented:
@dashea can you resolve the conflicts so we can merge ?

atodorov commented:
![](https://github.trello.services/images/mini-trello-icon.png) [(3) [QE, API] - enable static analysis for Rust code](https://trello.com/c/Xw9jVsEh/231-3-qe-api-enable-static-analysis-for-rust-code)

dashea commented:
Rebased to master, add the #![cfg_attr(...)] stuff to ignore the rocket + pass-by-value, and added some new changes for the test code that hadn't been getting been getting linted.
